### PR TITLE
chore(app): replace custom properties on req with res.locals

### LINKED ISSUE #379: [REF] Property setting on express request
## Problem
Express request objects are not meant to have properties set on them (`req.property = some_value`), and the types reflect this (no way to adequately reflect that `req` has a property set). 

## Solution
We could adopt either
1. declaration merging ([like so](https://dev.to/kwabenberko/extend-express-s-request-object-with-typescript-declaration-merging-1nn5))
2. set properties on `res.locals` for all intents identical apart from naming w/ built in support from express. 

The suggested solution is to use `res.locals` as different properties set using declaration merging will all be merged into 1 giant request type, which would be extremely confusing.

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Application code previously used to set custom properties on the `req` object of express. This is acceptable in javascript but as express typings doesn't expose such properties on the request object, will prove to be troublesome to type in typescript.

Express instead has a [`res.locals` ](https://expressjs.com/en/api.html#res.locals) property (and a corresponding type parameter in the `RequestHandler` type) that is used for such custom properties. We should instead be using this for ease of transition to typescript.

Part of #403
Closes #379 

## Solution
1. change existing declarations of `req.property = some_value` into `res.locals.property = some_value`
2. update call sites of `req.property` to `res.locals.property`

### CODE REVIEW

**Review by alexanderleegs — APPROVED** [review]
lgtm!

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
We missed changing a few endpoints -
- newroutes/authenticated/sites
- newmiddleware/auth

**Comment by seaerchin** [comment]
> We missed changing a few endpoints -
> 
> * newroutes/authenticated/sites
> * newmiddleware/auth

good catch! missed this because i forgot that `.` doesn't include line breaks, which resulted in a faulty regex used for this

File: newmiddleware/auth.js [hunk 1]
```diff
@@ -32,11 +32,10 @@ class AuthMiddleware {
   // Replace access token with site access token if it is available
   async useSiteAccessTokenIfAvailable(req, res, next) {
     const {
-      accessToken: userAccessToken,
       params: { siteName },
     } = req
     const {
-      locals: { userId },
+      locals: { userId, accessToken: userAccessToken },
     } = res
```
Comment (alexanderleegs) [inline newmiddleware/auth.js:39]: Minor nit: could we swap this to
```
const { siteName } = req.params
const { userId, accessToken: userAccessToken } = res.locals
```
?

Same for newroutes/authenticated/sites!

**Review by alexanderleegs — APPROVED** [review]
Minor nit, but lgtm otherwise!

File: newmiddleware/auth.js (hunk 1 above)
Comment (seaerchin) [reply newmiddleware/auth.js:39]: good catch!! i think this helps with readability and i'd be more than happy to make this change!

**Pull request merged**